linc_convert/utils/zarr/generate_pyramid.py

```diff
@@ -0,0 +1,173 @@
+import itertools
+import re
+from typing import Any, Literal
+
+import nibabel as nib
+import numpy as np
+import zarr
+
+from linc_convert.utils.math import ceildiv
+from linc_convert.utils.unit import convert_unit
+
+
+def generate_pyramid(
+    omz: zarr.Group,
+    levels: int | None = None,
+    ndim: int = 3,
+    max_load: int = 512,
+    mode: Literal["mean", "median"] = "median",
+    no_pyramid_axis: int | str | None = None,
```

Comment:
Changing type based on how the argument is used:

```
for i, length in enumerate(prev_shape):
    if i == no_pyramid_axis:
        ...
```
```suggestion
    no_pyramid_axis: int | None = None,
```